### PR TITLE
promotions-controller

### PR DESCRIPTION
**Pull request by tatitati**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/FindByPid/TlecController.php [hunk 1]
```diff
@@ -58,6 +65,8 @@ public function __invoke(
             'clips' => $clips,
             'galleries' => $galleries,
             'mapPresenter' => $mapPresenter,
+            'regularPromotions' => $promotions['regular'],
```
Comment (BPScott) [inline src/Controller/FindByPid/TlecController.php]: The list of promos shall be displayed in one long list, so there's not much point passing both arrays in separately. Check current site to see if the regular or super promos go first.

Also have you confirmed that the promos shall be output in the correct order?

File: src/Controller/FindByPid/TlecController.php [hunk 2]
```diff
@@ -42,6 +44,11 @@ public function __invoke(
             $galleries = $aggregationService->findDescendantGalleries($programme, 4);
         }
 
+        $promotions = $promotionService->findActivePromotionsByEntity($programme);
+        /**
+         * @todo: get related links for each promo
```
Comment (BPScott) [inline src/Controller/FindByPid/TlecController.php]: This shall be done as part of the service level, so this is the wrong place for this todo statement.

File: src/Controller/FindByPid/TlecController.php (hunk 1 above)
Comment (tatitati) [inline src/Controller/FindByPid/TlecController.php]: Yeah. I can confirm that the order is the weight. No the type of promotion.
is_descendants | weight
0	|    1
....
1	|    8
0	|    9
0	|    10
1	|    11
0	|    12
1	|    13
0	|    14
0	|    15

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**